src/perform_step/low_order.jl

```diff
@@ -104,6 +104,44 @@ end
   @.. u = uprev + dt * rtmp
 end
 
+# weak approximation EM
+@muladd function perform_step!(integrator,cache::SimplifiedEMConstantCache,f=integrator.f)
+  @unpack t,dt,uprev,u,W,p = integrator
+
+  K = uprev .+ dt .* integrator.f(uprev,p,t)
+
+  _dW = map(x -> sign(x) > 0.0 ? integrator.sqdt : -integrator.sqdt, W.dW)
+  if !is_diagonal_noise(integrator.sol.prob) || typeof(W.dW) <: Number
+    noise = integrator.g(uprev,p,t)*_dW
+  else
+    noise = integrator.g(uprev,p,t).*_dW
+  end
+
+  u = K + noise
+  integrator.u = u
+end
+
+@muladd function perform_step!(integrator,cache::SimplifiedEMCache,f=integrator.f)
+  @unpack rtmp1,rtmp2 = cache
+  @unpack t,dt,uprev,u,W,p = integrator
+
+  integrator.f(rtmp1,uprev,p,t)
+
+  @.. u = uprev + dt * rtmp1
+
+  integrator.g(rtmp2,uprev,p,t)
+
+  _dW = map(x -> sign(x) > 0.0 ? integrator.sqdt : -integrator.sqdt, W.dW)
```

Comment:
the cache version should be non-allocating, so make a `_dW` cache and then fill it with a `map!`. Pull the anonymous function outside of the function to be some generic function so it precompiles.

Reply:
Okay! I also already thought about it but I wasn't sure. Should I put the function in a separate weak_utils.jl file? I guess the two-point and three-point distributed random variables will be used in several places for the next solvers

Reply:
>Okay! I also already thought about it but I wasn't sure. Should I put the function in a separate weak_utils.jl file? I guess the two-point and three-point distributed random variables will be used in several places for the next solvers

Yup, makes complete sense.